src/helpers/augmentedReality/downloadObject.js

```diff
@@ -42,6 +45,21 @@ export const downloadObject = async ({ index, data, setData }) => {
       );
 
       try {
+        /*
+        since light doesn't have a downloadable url, a control 
+        has been added here to save it to JSON
+        */
+        if (type === 'light') {
+          downloadedData[index].payload.scenes[sceneIndex].localUris?.push({
```

Comment:
why are thee values pushed apart from the other values at

```
downloadedData[index].payload.scenes[sceneIndex].localUris?.push({
  chromaKeyFilteredVideo, // HEX Color Code
  id,
  isSpatialSound, // Boolean
  maxDistance: parseFloat(maxDistance),
  minDistance: parseFloat(minDistance),
  physicalWidth: parseFloat(physicalWidth),
  position, // Array [x,y,z]
  rolloffModel, // String (none, linear, or logarithmic)
  rotation, // Array [x,y,z]
  scale, // Array [x,y,z]
  size,
  title,
  type,
  uri: fileSystemDownload.uri
});
```
?

what happens if `color`, `intensity` and `temperature` would be added there as well?

Reply:
The reason for keeping the props separate is that the prop with light type has no uri. Since there is no uri, `downloadResumable` will give an error. So it will go to catch blog before adding to `localUris` and these props will not be added.

Reply:
https://github.com/ikuseiGmbH/smart-village-app-app/pull/674/commits/94f2c4db4032096146eea25f4c89bf566ce7a02a